sql/dj_setup_database.in

```diff
@@ -354,6 +370,64 @@ upgrade)
 	verbose "DOMjudge database upgrade completed."
 	;;
 
+snapshot-dump)
+	read_dbpasswords
+	if [ -z "$DUMPNAME" ]; then
+		usage
+		exit 1
+	fi
+
+	if [ -f "${DATABASEDUMPDIR}/${DUMPNAME}.sql.gz" ]; then
+		while true; do
+			read -p "Overwrite existing database dump (y/N)? " yn
+			case $yn in
+				[Yy]* ) create_database_dump $DUMPNAME ;;
+				''|[Nn]* ) break;;
+			esac
+		done
+	else
+		create_database_dump $DUMPNAME
+	fi
+
+	read -p "Provide prefix or say 'no' to abort. Format: [prefix]-$1 " prefix
+	case $prefix in
+		''|[Nn]o|[Nn] ) exit 2; break;;
+		* ) create_database_dump ${prefix}-${DUMPNAME}; break;;
+	esac
+	;;
+
+snapshot-load)
+	choice=""
```

Comment:
Can we also support providing this as a command line option, so that this is scriptable?

Reply:
That's why we have the `-d` option now. Or `load <init-db>`.